challenges/test/lib/src/main/java/code/stacksandqueues/Queue.java

```diff
@@ -0,0 +1,50 @@
+package code.stacksandqueues;
+
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+
+public class Queue<T> {
+  private Node<T> front;//this will point to the begnning
+  private Node<T> back;//this will point essentially to the end
+
+  public void enqueue(T value){
+    Node<T> newNode = new Node<>(value);//using modified node
+    //so if the whole thing is empty, we will set the node to start the whole line
+    if(this.isEmpty()){
```

Comment:
You did, this.isEmpty() here, but the rest of the methods you did front==null? Works either way!